src/webgpu/api/validation/resource_usages/texture/in_render_common.spec.ts

```diff
@@ -95,3 +101,128 @@ g.test('subresources_from_same_texture_as_color_attachments')
+      )
+      .combine('inSamePass', [true, false])
+  )
+  .fn(async t => {
+    const {
+      colorAttachmentBaseLevel,
+      colorAttachmentBaseLayer,
+      bindGroupViewBaseLevel,
+      bindGroupViewLevelCount,
+      bindGroupViewBaseLayer,
+      bindGroupViewLayerCount,
+      bindGroupUsage,
+      inSamePass,
+    } = t.params;
+
+    const bindGroupLayoutEntry: GPUBindGroupLayoutEntry = {
+      binding: 0,
+      visibility: GPUShaderStage.FRAGMENT,
+    };
+    let textureUsage = GPUTextureUsage.RENDER_ATTACHMENT;
+    const viewDimension = bindGroupViewLayerCount > 1 ? '2d-array' : '2d';
+    switch (bindGroupUsage) {
+      case 'texture':
+        bindGroupLayoutEntry.texture = { viewDimension };
+        textureUsage |= GPUTextureUsage.TEXTURE_BINDING;
+        break;
+      case 'storage':
+        bindGroupLayoutEntry.storageTexture = {
+          access: 'write-only',
+          format: 'rgba8unorm',
+          viewDimension,
+        };
+        textureUsage |= GPUTextureUsage.STORAGE_BINDING;
+        break;
+      default:
+        unreachable();
+        break;
+    }
+
+    const texture = t.device.createTexture({
+      format: 'rgba8unorm',
+      usage: textureUsage,
+      size: [kTextureSize, kTextureSize, kTextureLayers],
+      mipLevelCount: kTextureLevels,
+    });
+    const bindGroupView = texture.createView({
+      dimension: bindGroupViewLayerCount > 1 ? '2d-array' : '2d',
+      baseArrayLayer: bindGroupViewBaseLayer,
+      arrayLayerCount: bindGroupViewLayerCount,
+      baseMipLevel: bindGroupViewBaseLevel,
+      mipLevelCount: bindGroupViewLevelCount,
+    });
+
+    const bindGroupLayout = t.device.createBindGroupLayout({
+      entries: [bindGroupLayoutEntry],
+    });
+    const bindGroup = t.device.createBindGroup({
+      layout: bindGroupLayout,
+      entries: [{ binding: 0, resource: bindGroupView }],
+    });
+
+    const colorAttachment = t.getColorAttachment(texture, {
+      baseArrayLayer: colorAttachmentBaseLayer,
+      arrayLayerCount: 1,
+      baseMipLevel: colorAttachmentBaseLevel,
+      mipLevelCount: 1,
+    });
+
+    const encoder = t.device.createCommandEncoder();
+    const renderPass = encoder.beginRenderPass({
+      colorAttachments: [colorAttachment],
+    });
+    if (inSamePass) {
+      renderPass.setBindGroup(0, bindGroup);
+      renderPass.end();
+    } else {
+      renderPass.end();
+
+      const texture2 = t.device.createTexture({
+        format: 'rgba8unorm',
+        usage: GPUTextureUsage.RENDER_ATTACHMENT,
+        size: [kTextureSize, kTextureSize, 1],
+        mipLevelCount: 1,
+      });
+      const colorAttachment2 = t.getColorAttachment(texture2);
+      const renderPass2 = encoder.beginRenderPass({
+        colorAttachments: [colorAttachment2],
+      });
+      renderPass2.setBindGroup(0, bindGroup);
+      renderPass2.end();
+    }
+
+    const colorAttachmentBaseLevelSuccess =
```

Comment:
nit: colorAttachmentLevelDisjoint or similar. "success" makes it sound like
`colorAttachmentBaseLevelSuccess || colorAttachmentBaseLayerSuccess` should be
`colorAttachmentBaseLevelSuccess && colorAttachmentBaseLayerSuccess`, which would be wrong (I almost suggested that change).